lib/sablon/parser/mail_merge.rb

```diff
@@ -111,13 +111,16 @@ def parse_fields(xml)
       end
 
       private
+
       def build_complex_field(node)
         possible_field_node = node.parent
         field_nodes = [possible_field_node]
         while possible_field_node && possible_field_node.search(".//w:fldChar[@w:fldCharType='end']").empty?
           possible_field_node = possible_field_node.next_element
           field_nodes << possible_field_node
         end
+        # removing the nil value that gets appended if no end node is found
+        field_nodes.pop if field_nodes.last.nil?
         ComplexField.new(field_nodes)
```

Comment:
This is still creating a `ComplexField` instance for the field we don't have an end tag for. Do you think we could skip instantiation all together. Otherwise we still have half-completed objects in the graph.